core/server/master/src/main/java/alluxio/master/meta/AlluxioMasterRestServiceHandler.java

```diff
@@ -126,23 +197,681 @@ public Response getInfo(@QueryParam(QUERY_RAW_CONFIGURATION) final Boolean rawCo
+          }
+          String endParam = requestEnd;
+          // If no param "end" presents, the offset is relative to the beginning; otherwise, it is
+          // relative to the end of the file.
+          if (endParam == null) {
+            offset = relativeOffset;
+          } else {
+            offset = fileInfo.getLength() - relativeOffset;
+          }
+          if (offset < 0) {
+            offset = 0;
+          } else if (offset > fileInfo.getLength()) {
+            offset = fileInfo.getLength();
+          }
+          try {
+            AlluxioURI path = new AlluxioURI(currentFileInfo.getAbsolutePath());
+            FileSystem fs = FileSystem.Factory.get();
+            URIStatus status = fs.getStatus(path);
+            if (status.isCompleted()) {
+              OpenFileOptions options = OpenFileOptions.defaults().setReadType(ReadType.NO_CACHE);
+              try (FileInStream is = fs.openFile(path, options)) {
+                int len = (int) Math.min(5 * Constants.KB, status.getLength() - offset);
+                byte[] data = new byte[len];
+                long skipped = is.skip(offset);
+                if (skipped < 0) {
+                  // nothing was skipped
+                  fileData = "Unable to traverse to offset; is file empty?";
+                } else if (skipped < offset) {
+                  // couldn't skip all the way to offset
+                  fileData = "Unable to traverse to offset; is offset larger than the file?";
+                } else {
+                  // read may not read up to len, so only convert what was read
+                  int read = is.read(data, 0, len);
+                  if (read < 0) {
+                    // stream couldn't read anything, skip went to EOF?
+                    fileData = "Unable to read file";
+                  } else {
+                    fileData = WebUtils.convertByteArrayToStringWithoutEscape(data, 0, read);
+                  }
+                }
+              }
+            } else {
+              fileData = "The requested file is not complete yet.";
+            }
+            List<UIFileBlockInfo> uiBlockInfo = new ArrayList<>();
+            for (FileBlockInfo fileBlockInfo : mMasterProcess.getMaster(FileSystemMaster.class)
+                .getFileBlockInfoList(path)) {
+              uiBlockInfo.add(new UIFileBlockInfo(fileBlockInfo));
+            }
+            fileBlocks = uiBlockInfo;
+            highestTierAlias =
+                mMasterProcess.getMaster(BlockMaster.class).getGlobalStorageTierAssoc().getAlias(0);
+          } catch (AlluxioException e) {
+            fatalError = "Error: Fatal Error " + e.getMessage();
+          }
+          viewingOffset = offset;
+        }
+
+        AlluxioURI path = new AlluxioURI(currentFileInfo.getAbsolutePath());
+        if (path.isRoot()) {
+          pathInfos = new UIFileInfo[0];
+        } else {
+          String[] splitPath = PathUtils.getPathComponents(path.toString());
```

Comment:
Need to init `pathInfos` in the else clause, ie. `pathInfos = new UIFileInfo[splitPath.length]`